cpp/oneapi/dal/algo/kmeans_init/backend/gpu/compute_kernel_dense_dpc.cpp

```diff
@@ -41,54 +41,84 @@ using daal_kmeans_init_dense_batch_ucapi_kernel_t =
                                                                 Float>;
 
 template <typename Float>
-struct compute_kernel_gpu<Float, method::dense> {
-    compute_result operator()(const dal::backend::context_gpu& ctx,
-                              const descriptor_base& params,
-                              const compute_input& input) const {
-        auto& queue = ctx.get_queue();
-        interop::execution_context_guard guard(queue);
-
-        const auto data = input.get_data();
-
-        const int64_t column_count  = data.get_column_count();
-        const int64_t cluster_count = params.get_cluster_count();
-
-        daal_kmeans_init::Parameter par(cluster_count);
-
-        auto arr_data        = row_accessor<const Float>{ data }.pull(queue);
-        const auto daal_data = interop::convert_to_daal_sycl_homogen_table(queue,
-                                                                           arr_data,
-                                                                           data.get_row_count(),
-                                                                           data.get_column_count());
-
-        array<Float> arr_centroids = array<Float>::empty(queue, cluster_count * column_count);
-        const auto daal_centroids  = interop::convert_to_daal_sycl_homogen_table(queue,
-                                                                                arr_centroids,
-                                                                                cluster_count,
-                                                                                column_count);
-
-        const size_t len_daal_input                                       = 1;
-        daal::data_management::NumericTable* daal_input[len_daal_input]   = { daal_data.get() };
-        const size_t len_daal_output                                      = 1;
-        daal::data_management::NumericTable* daal_output[len_daal_output] = {
-            daal_centroids.get()
-        };
+using daal_kmeans_init_random_dense_batch_ucapi_kernel_t =
+    daal_kmeans_init::internal::KMeansInitDenseBatchKernelUCAPI<daal_kmeans_init::randomDense,
+                                                                Float>;
+
+template <typename Float>
+using daal_kmeans_init_plus_plus_dense_batch_ucapi_kernel_t =
+    daal_kmeans_init::internal::KMeansInitDenseBatchKernelUCAPI<daal_kmeans_init::plusPlusDense,
+                                                                Float>;
+
+template <typename Float>
+using daal_kmeans_init_parallel_plus_dense_batch_ucapi_kernel_t =
+    daal_kmeans_init::internal::KMeansInitDenseBatchKernelUCAPI<daal_kmeans_init::parallelPlusDense,
+                                                                Float>;
+
+template <typename Float, typename Method>
+compute_result compute_kernel_gpu<Float, Method>::operator()(const dal::backend::context_gpu& ctx,
```

Comment:
Implementation is inconsistent with CPU branch. In CPU branch you have method `call_daal_kernel`, but here you call DAAL kernel from operator.

- Introduce `call_daal_kernel`

Reply:
done